csvcubed/tests/behaviour/steps/qbwriter.py

```diff
@@ -735,3 +743,89 @@ def step_impl(context, cube_name: str):
 def step_impl(context):
     rdf_to_add = context.text
     context.turtle += rdf_to_add
+
+
+@Then("the cube's metadata should contain URLs with file endings")
+def step_impl(context):
+    temp_dir = get_context_temp_dir_path(context)
+    assertURIStyle(URIStyle.Standard, temp_dir, context.csv_file_name)
+
+
+@Then("the cube's metadata should contain URLs without file endings")
+def step_impl(context):
+    temp_dir = get_context_temp_dir_path(context)
+    assertURIStyle(URIStyle.WithoutFileExtensions, temp_dir, context.csv_file_name)
+
+
+@Given(
+    'a single-measure QbCube named "{cube_name}" configured with "{uri_style}" URI style'
+)
+def step_impl(context, cube_name: str, uri_style: str):
+    context.cube = _get_single_measure_cube_with_name_and_id(
+        cube_name, None, URIStyle[uri_style]
+    )
+
+
+def assertURIStyle(uri_style: URIStyle, temp_dir: Path, csv_file_name: str):
+    metadataFilePath = temp_dir.joinpath(f"{csv_file_name}-metadata.json")
+    g = Graph()
+    g.parse(metadataFilePath)
+
+    predicate_whitelist = [
```

Comment:
I see what you've done here but I do worry these whitelists will make the tests fairly brittle if and when we add new RDF into the CSV-Ws in future. 

I think the way I would have approached this problem is to look for *relative URIs* which are defined as part of the graph - these are the new resources we're creating which use the newly generated URIs.

RDFlib isn't happy with relative URIs so it converts them to absolute URIs using the `metadataFilePath` in this instance. Now we've had some problems with the way that rdflib converts file paths into URIs (where Jenkins puts things like `@2` in folder names/paths), so I'd recommend setting `publicID="http://prefix-for-relative-uris"` in your call to `g.parse` which would mean that all of your relative URIs would end up with `http://prefix-for-relative-uris` prepended. Then you can filter by URIs that begin with that and ensure that they all either have or don't have the `.csv` extension on the end.

-------

Btw, `metadataFilePath` betrays your allegiance for other languages ;) would be `metadata_file_path` by convention in python.

Reply:
ah damn i missed one - was a few days in when i noticed the snake casing

Reply:
ok will switch to that approach. I noticed it was the relative ones I wanted but it felt a bit dirty looking for what rdflib happend to convert them too. Using publicID makes it a bit better.